pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/InTransformFunction.java

```diff
@@ -365,11 +380,56 @@ public int[] transformToIntValuesSV(ValueBlock valueBlock) {
             }
           }
           break;
+        case UNKNOWN:
+          break;
         default:
           throw new IllegalStateException();
       }
     }
 
     return _intValuesSV;
   }
+
```

Comment:
Can we also implement transformToInValuesSVWithNull for perf reason?

Reply:
I suggest not to implement `transformToInValuesSVWithNull`.

- Implementing something like [`BaseTransformFunction.transformToInValuesSVWithNull`](https://github.com/apache/pinot/blob/13a792f03f267bf53e36fc06340d43d8fc1d7af4/pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java#L184) violates [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
- There wouldn't be much performance gain because `BaseTransformFunction.transformToInValuesSVWithNull` still gets int[] and bitmap separately.

Reply:
Performance matters more here than DRY. :) 

Overriding transformToInValuesSVWithNull will avoid calling the base function's transformToInValuesSVWithNull and you can call _mainFunction.transformToInValuesSVWithNull to get the values and null bit map at the same time.

Reply:
Had an offline discussion with @61yao 

- There is performance improvement if we override `transformToInValuesSVWithNull` because 
   - Some subclass's `transformToInValuesSVWithNull` is faster because they can compute the values and null map together (e.g. CASE WHEN).
   - `BaseTransformFunction.transformToInValuesSVWithNull` calls `transformToInValuesSVWithNull`, which can be a subclass (polymorphism) and is faster (explained in the above sub-point).
- The `InTransformFunction.transformToInValuesSV` is long, if we override `transformToInValuesSVWithNull` with a similar implementation, the repeated code decreases maintainability. Added a TODO and we will override after [refactoring](https://github.com/apache/pinot/issues/10618).

Reply:
The problem without implementing transformToXXWithNull is that we cannot fill in the default values for all nulls.
For example, if the value is later evaluated null in getNullBitmap call, the intValue field is not updated.
I think it is fine now because filling default value rule is not strictly followed in default implementation. We can change this later when default implementation is fixed.

Do you mind putting a TODO or a comment noting that the value for null is undefined/indeterministic if the transformToInt API is called?